latte/dataset/cars3d.py

```diff
@@ -0,0 +1,137 @@
+""" Script for downloading and loading the Cars3D dataset.
+    Some parts based on disentanglement_lib (https://github.com/google-research/disentanglement_lib). """
+
+import dataclasses
+import pathlib
+import os
+import tarfile
+import shutil
+
+from typing import Union
+
+from PIL import Image
+import h5py
+import numpy as np
+import scipy.io as sio
+
+from latte.dataset import utils as dsutils
+
+
+DEFAULT_TARGET = "dataset/raw"
+CARS3D_URL = "http://www.scottreed.info/files/nips2015-analogy-data.tar.gz"
+MD5_CHECKSUM = "4e866a7919c1beedf53964e6f7a23686"
+
+
+@dataclasses.dataclass
+class Cars3DDataset:
+    imgs: np.ndarray
+    latents_values: np.ndarray
+
+
+def prepare_files(filepath: pathlib.Path) -> None:
+    """Function extracting and preparing the Cars3D data.
+
+    Args:
+        filepath: directory for storing the temporary data and the dataset
+
+    Returns:
+        Cars3DDataset object
+    """
+
+    archive_filepath = filepath / "cars3d.tar.gz"
```

Comment:
As described below in more detail, in the future you may consider the https://docs.python.org/3/library/tempfile.html :slightly_smiling_face: